### ISSUE TITLE
Allow partial enablement of modules

### ISSUE THREAD

**Issue by hahnjo** [root-project/root#12591]
Commit 46ba2eefd00371b024ae239626c7600cd3259885 (https://github.com/root-project/root/pull/12082) enabled a few modules for Windows, but added `NO_CXXMODULE` to a couple others that don't work yet. This requires support for figuring out which dependencies have modules and which ones don't. Unfortunately, this support had some flaws and had to be reverted in commit 2d7373d47ef2acc53f4eed004082fee83caab6a3 (https://github.com/root-project/root/pull/12561) - see the summary and commit messages for more details.

To summarize, the core of the problem is that the build system expected a dependency to have a module if it had been registered beforehand. This is incompatible with how we build `libCore` and some other fundamental libraries. We likely have to implement it the other way around, ie keep track of which targets will produce a module. This may be enough and work for the use cases we have on Windows.
